.github/workflows/upload-and-deploy.yaml

```diff
@@ -46,13 +48,171 @@ jobs:
+    environment: develop
+    steps:
+      - name: Assume AWS role
+        uses: aws-actions/configure-aws-credentials@v2
+        with:
+          role-to-assume: ${{ vars.AWS_CREDENTIALS_IAM_ROLE }}
+          aws-region: "us-east-1"
+          # unmasking of the AWS account ID allows the acct id to pass through outputs
+          mask-aws-account-id: "no"
+
+      - name: Login to Amazon ECR
+        id: login-ecr
+        uses: aws-actions/amazon-ecr-login@v1
+
+      - name: Get ECR secret names
+        id: ecr
+        run: |
+          usernameKey=docker_username_$(echo ${{ steps.login-ecr.outputs.registry }} | tr '.-' _)
+          echo "username-key=$usernameKey" >> $GITHUB_OUTPUT
+          passwordKey=docker_password_$(echo ${{ steps.login-ecr.outputs.registry }} | tr '.-' _)
+          echo "password-key=$passwordKey" >> $GITHUB_OUTPUT
+      - uses: actions/checkout@v3
+
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v2
+
+      - name: Build and push to ECR
+        id: docker-build-push
+        uses: docker/build-push-action@v4
+        with:
+          push: true
+          tags: ${{ steps.login-ecr.outputs.registry }}/pytest:${{ github.ref_name }}_${{ matrix.tag_name }}
+          file: ${{ matrix.dockerfile }}
+          cache-from: type=local,src=/tmp/.buildx-cache
+          cache-to: type=local,dest=/tmp/.buildx-cache
+    outputs:
+      ecr-registry: ${{ steps.login-ecr.outputs.registry }}
+      ecr-username: ${{ steps.login-ecr.outputs[steps.ecr.outputs.username-key] }}
+      ecr-password: ${{ steps.login-ecr.outputs[steps.ecr.outputs.password-key] }}
+
+
+  glue-unit-tests:
+    name: Run Pytest unit tests for AWS glue
+    needs: pytest-docker
+    environment: develop
+    runs-on: ubuntu-latest
+    # These permissions are needed to interact with GitHub's OIDC Token endpoint.
+    permissions:
+      id-token: write
+      contents: read
+    strategy:
+      matrix:
+          tag_name:
+            ["aws_glue_3", "aws_glue_4"]
+    container:
+      image: ${{ needs.pytest-docker.outputs.ecr-registry }}/pytest:${{ github.ref_name }}_${{ matrix.tag_name }}
+      credentials:
+        username: ${{ needs.pytest-docker.outputs.ecr-username }}
+        password: ${{ needs.pytest-docker.outputs.ecr-password }}
+      env:
+        DISABLE_SSL: true
+      options: "--user root"
+    steps:
+      - name: Assume AWS role
+        uses: aws-actions/configure-aws-credentials@v2
+        with:
+          role-to-assume: ${{ vars.AWS_CREDENTIALS_IAM_ROLE }}
+          aws-region: "us-east-1"
+      - uses: actions/checkout@v3
+      - run: chown -R glue_user $GITHUB_WORKSPACE
+      - run: su - glue_user --command "aws configure set aws_access_key_id $AWS_ACCESS_KEY_ID"
+      - run: su - glue_user --command "aws configure set aws_secret_access_key $AWS_SECRET_ACCESS_KEY"
+      - run: su - glue_user --command "aws configure set aws_session_token $AWS_SESSION_TOKEN"
```

Comment:
yeah, I'm not 100% sure if these lines are required as I copied them over from BD. When I looked up setting up ECR and using the containers from ECR via github actions, it seemed very particular about things, so I just wanted to keep it consistent with BD here since it's working there.